src/com/mendix/widget/ProgressBar/ProgressBar.ts

```diff
@@ -0,0 +1,96 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+import { createElement } from "react";
+import { render, unmountComponentAtNode } from "react-dom";
+
+import {
+    BarType,
+    BootstrapStyle,
+    ProgressBar as ProgressBarComponent,
+    ProgressBarProps
+} from "./components/ProgressBar";
+
+class ProgressBar extends WidgetBase {
+    // Parameters configured from modeler
+    progressAttribute: string;
+    bootstrapStyleAttribute: string;
+    progressStyle: BootstrapStyle;
+    barType: BarType;
+    textColorSwitch: number;
+    maximumValueAttribute: string;
+    onclickMicroflow: string;
+    // Internal variables
+    private contextObject: mendix.lib.MxObject;
+
+    update(object: mendix.lib.MxObject, callback?: Function) {
+        this.contextObject = object;
+        this.resetSubscriptions();
+        this.updateRendering();
+
+        if (callback) {
+            callback();
+        }
+    }
+
+    uninitialize(): boolean {
+        unmountComponentAtNode(this.domNode);
+
+        return true;
+    }
+
+    private updateRendering() {
+        render(createElement(ProgressBarComponent, this.getProgressBarProps()), this.domNode);
+    }
+
+    private getProgressBarProps(): ProgressBarProps {
+        const { bootstrapStyleAttribute, contextObject, maximumValueAttribute, progressAttribute } = this;
+        let progress = 0;
+        let bootstrapStyle = this.progressStyle;
+        let maximumValue = 100;
+        if (this.contextObject) {
+            progress = Math.round(parseInt(contextObject.get(progressAttribute) as string, 10));
+            bootstrapStyle = contextObject.get(bootstrapStyleAttribute || "") as BootstrapStyle || bootstrapStyle;
```

Comment:
If bootstrapStypeAttribute is not set, it passes in "" which lets the function return undefined/null.
It's a shortcut & more accurate way of writing
```
bootstrapStyle = boostrapStypeAttribute
    ? contextObject.get(bootstrapAttribute) as BootstrapStyle
    : bootstrapStyle
```